pred.py

```diff
@@ -2,98 +2,111 @@
 from datasets import load_dataset
 import torch
 import json
-from transformers import AutoTokenizer, LlamaTokenizer, LlamaForCausalLM, AutoModelForCausalLM
+from transformers import AutoConfig, AutoTokenizer, LlamaTokenizer, LlamaForCausalLM, AutoModelForCausalLM
 from tqdm import tqdm
 import numpy as np
 import random
 import argparse
 from llama_flash_attn_monkey_patch import replace_llama_attn_with_flash_attn
 import torch.distributed as dist
 import torch.multiprocessing as mp
+from transformers import (DynamicCache,
+                          SinkCache,
+                          SlidingWindowCache,
+                          QuantoQuantizedCache,
+                          )
+import gc
+from vllm import LLM, SamplingParams
 
 def parse_args(args=None):
     parser = argparse.ArgumentParser()
-    parser.add_argument('--model', type=str, default=None, choices=["llama2-7b-chat-4k", "longchat-v1.5-7b-32k", "xgen-7b-8k", "internlm-7b-8k", "chatglm2-6b", "chatglm2-6b-32k", "chatglm3-6b-32k", "vicuna-v1.5-7b-16k", "llama3.1-8b-instruct", "baselineWithCuratedFTLongMSL8K", "baselineWithCuratedFTLongMSL8KCQ"])
+    parser.add_argument('--model', type=str, default=None,)
+    parser.add_argument('--max_seq_len', type=int, default=7500,)
     parser.add_argument('--e', action='store_true', help="Evaluate on LongBench-E")
+    parser.add_argument('--kv_method', type=str, default='full',)
+    parser.add_argument('--prefill_kv', action='store_true', help="prefill kv cache before generate function")
+    parser.add_argument('--backend', type=str, default='hf',choices=["hf","vllm"])
+    parser.add_argument('--load_in_8bit', action='store_true')
     return parser.parse_args(args)
 
-# This is the customized building prompt for chat models
```

Comment:
Why removed entirely? I see you also removed the special cases for different models in lines 58-71, what impact do they make on these models? I like the simplicity after clearing it out, but might we need this in the future for particular models and are we going to be still compatible with the original benchmark?